server-sdk/src/main/java/com/amazonaws/ml/mms/servingsdk/ModelServerEndpoint.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
+ * with the License. A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
+ * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
+ * and limitations under the License.
+ */
+package com.amazonaws.ml.mms.servingsdk;
+
+import com.amazonaws.ml.mms.servingsdk.http.Request;
+import com.amazonaws.ml.mms.servingsdk.http.Response;
+
+
+public abstract class ModelServerEndpoint {
+    public void doGet(Request req, Response res, Context ctx) {
```

Comment:
I'm assuming you want to make this method mandatory to override by the child classes ? 
If that's the case, then we can mark this method also as abstract, thus enforcing the downstream child classes to have an impl for it.

Reply:
The idea here is that if customers don't override, we throw exception. Each of these endpoints will have implementation for all the HTTP methods. Not all are used by customers.